### DISCUSSION TITLE
A Question About Thing Deletion

### DISCUSSION THREAD

**Discussion by fix-all** [eclipse-ditto/ditto#1917] [General]
hi,
I used the Ditto-client to delete a Thing using the twin.delete method. After deletion, when I use /api/2/search/things?filter=eq(thingId,"test:test190423043997696") to query, the result is empty. However, when I use /api/2/things/test:test190423043997696, the Thing still exists. How should I handle this situation?

**Comment by thjaeckle** [comment]
This should not be the case.
Could you also reproduce when deleting via the HTTP DELETE endpoint?

**Reply by fix-all** [reply]
The reason for this phenomenon is that the data in the things-search database is partially less than the data in the things database.
  I first made a query which queried the things-search database and then deleted the data based on the queried ThingId so that the part of the data that was less was not deleted.
  But I'm still not sure why things-search is missing a part of the data
